_source/_posts/2022-01-05-eleventy.md

```diff
@@ -0,0 +1,375 @@
+```
+
+Then you can run the command `npx @11ty/eleventy --serve`.  By default 11ty will attempt to use port `8080` on localhost, so navigate to `http://localhost:8080` and you can see your **Hello World** page.  A quick note here, if you are using Windows you will need to open the index file you just created and resave it with `UTF-8` encoding as 11ty uses `UTF-16 LE` by default.  
+
+## Add a Couple Static Pages
+
+Now you can add some content to your site.  Your site will consist of an index page to greet users, a login page, a readme to demonstrate the usage of markdown, and a page that is under authentication.  
+
+First, open the `index.html` file you just created and replace the code with the following.
+
+```HTML
+---
+layout: _layout_unauthenticated.11ty.js
+---
+
+<div class="row align-items-md-stretch">
+    <div class="col-md-12">
+        <div class="h-100 p-5 text-white bg-secondary rounded-3">
+            <h2>Using Okta with 11ty</h2>
+            <p>A small tutorial on writing static sites with 
+                <a class="text-dark" href="https://www.11ty.dev/" target="_blank">11ty</a>, 
+                secured by <a class="text-dark" href="https://www.okta.com/" target="_blank">Okta</a>
+                written by <a class="text-dark" href="https://profile.fishbowlllc.com/" target="_blank">Nik Fisher.</a>
+            </p>
+            <a class="btn btn-outline-light" role="button" href="Login">Login Now</a>
+            <a class="btn btn-light" role="button" href="README">README</a>
+        </div>
+    </div>
+</div>
+```
+
+You will notice the `layout` variable at the top of the.  You will add your layout pages later in this tutorial but take notes that you will need to assign a layout to each page if you wish.
+
+Next, create a file for `login.html` and add the following code.  
+
+```HTML
+---
+layout: _layout_unauthenticated.11ty.js
+---
+
+<div id="okta-login-container"></div>
+<script>
+  oktaSignIn.authClient.token.getUserInfo().then(function(user) {
+    window.location='/protected';
+  },function(error) {
+    oktaSignIn.showSignInToGetTokens({
+      el: '#okta-login-container'
+    }).then(function(tokens) {
+      oktaSignIn.authClient.tokenManager.setTokens(tokens);
+      oktaSignIn.remove();
+      window.location='/protected';
+    }).catch(function(err) {
+      console.error(err);
+    });
+  });
+</script>
+```
+
+This page has a bit more to it.  The only HTML here is a `div` with `id="okta-login-container"`.  You call the `OktaSignIn` features and display the login screen to the user in The javascript of this page.  Okta will take care of creating the login form and handling the actual authentication.  All you need to do is capture the token on the back and store the token.  Once you have done that you will redirect the user to the `protected` page.  
+
+You can add the protected page now.  Add the code below to it.
```

Comment:
Should the file name be `protected.html`?

Reply:
Why not?